mldsa/ntt.c

```diff
@@ -151,31 +189,25 @@ void ntt(int32_t a[MLDSA_N])
  *              MLDSA_Q in absolute value. Output coefficient are smaller than
  *MLDSA_Q in absolute value.
  *
- * Arguments:   - uint32_t p[MLDSA_N]: input/output coefficient array
+ * Arguments:   - uint32_t a[MLDSA_N]: input/output coefficient array
  **************************************************/
 void invntt_tomont(int32_t a[MLDSA_N])
+
 {
-  unsigned int start, len, j, k;
-  int32_t t, zeta;
+  unsigned int layer, j;
   const int32_t f = 41978; /* mont^2/256 */
 
-  k = 256;
-  for (len = 1; len < MLDSA_N; len <<= 1)
+  for (layer = 8; layer >= 1; layer--)
+  __loop__(
+    invariant(layer <= 8)
+    invariant(array_abs_bound(a, 0, MLDSA_N, (MLDSA_N >> layer) * MLDSA_Q)))
```

Comment:
Please add comment explaining invariant here as per @rod-chapman's version of the code

Reply:
added in [6c53ce6](https://github.com/pq-code-package/mldsa-native/pull/179/commits/6c53ce6c53e2d6a6ebc4c95087f4ab28b0d0708a)